<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Gost3410_12_MPW</class>
 <widget class="QWidget" name="Gost3410_12_MPW">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>855</width>
    <height>577</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Параметры эллиптической кривой</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Модуль &quot;p&quot; эллиптической кривой</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="elleptic_curve_p_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Коэффициент &quot;a&quot; эллиптической кривой</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="elleptic_curve_a_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Коэффициент &quot;b&quot; эллиптической кривой</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="elleptic_curve_b_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Порядок группы точек эллиптической кривой &quot;m&quot; </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="elleptic_curve_m_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Порядок циклической подгруппы группы точек эллиптической &quot;q&quot;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="elleptic_curve_q_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Ключи ЭП</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Ключ подписи d</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="sign_key_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Координата &quot;x&quot; ключа проверки подписи Q</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="signcheck_key_x_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Координата &quot;y&quot; ключа проверки подписи Q</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="signcheck_key_y_edit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="generate_sign_keys_button">
       <property name="text">
        <string>Сгенерировать ключи подписи и проверки подписи</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="save_keys_file_button">
         <property name="text">
          <string>Сохранить ключи</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="load_keys_file_button">
         <property name="text">
          <string>Загрузить ключи из файла</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Подпись</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="sign_file_button">
          <property name="text">
           <string>Подписать файл</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="signcheck_file_button">
          <property name="text">
           <string>Проверить подпись</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="task_label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Текущий процесс:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
